applications/tari_app_grpc/proto/types.proto

```diff
@@ -228,3 +228,37 @@ message ConsensusConstants {
     /// Block weight for kernels
     uint64 block_weight_kernels = 16;
 }
+
+
+
+message Address{
+    bytes address =1;
+    string last_seen = 2;
+    uint32 connection_attempts = 3;
+    uint32 rejected_message_count = 4;
+    uint64 avg_latency = 5;
+}
+
+message Peer{
+    /// Public key of the peer
+    bytes public_key =1;
+    /// NodeId of the peer
+    string node_id =2;
```

Comment:
```suggestion
    bytes node_id =2;
```

Reply:
You could also derive this from the public key to save a few bytes `NodeId::from_public_key`